modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4581,9 +4581,14 @@ public CodegenParameter fromParameter(Parameter parameter, Set<String> imports)
         String parameterModelName = null;
 
         if (parameter.getSchema() != null) {
-            parameterModelName = getParameterDataType(parameter ,parameter.getSchema());
-            parameterSchema = ModelUtils.getReferencedSchema(openAPI, parameter.getSchema());
```

Comment:
This changing results means that unaliasSchema is working differently than ModelUtils.getReferencedSchema for these parameter schemas.
Investigate the details of this.

Reply:
This is happening because if an enum schema is referenced, the schema with the reference is returned. This assumes that the generator generates a typed reference to that enum model.